gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/MultiRexster.java

```diff
@@ -0,0 +1,236 @@
+                        closeReason.set(null);
+
+                        logger.warn(ex.getMessage(), ex);
+                        currentGremlinContext.writeAndFlush(ex.getResponseMessage());
+                    }
+
+                    // work is done within the timeout period so cancel it
+                    cancelRequestTimeout();
+
+                    currentGremlinContext = queue.take();
+                }
+            } catch (Exception ex) {
+                stopAcceptingRequests();
+
+                // the current context gets its exception handled...
+                handleException(currentGremlinContext, ex);
+            }
+        } catch (RexsterException rexex) {
+            // remaining work items in the queue are ignored since this worker is closing. must send
+            // back some sort of response to satisfy the client. writeAndFlush code is different than
+            // the ResponseMessage as we don't want the message to be "final" for the Context. that
+            // status must be reserved for the message that caused the error
+            for (Context gctx : queue) {
+                gctx.writeAndFlush(ResponseStatusCode.PARTIAL_CONTENT, ResponseMessage.build(gctx.getRequestMessage())
+                        .code(ResponseStatusCode.SERVER_ERROR)
+                        .statusMessage(String.format(
+                                "An earlier request [%s] failed prior to this one having a chance to execute",
+                                currentGremlinContext.getRequestMessage().getRequestId())).create());
+            }
+
+            // exception should trigger a rollback in the session. a more focused rollback may have occurred
+            // during process() and the related result iteration IF transaction management was enabled on
+            // the request
+            closeTransactionSafely(Transaction.Status.ROLLBACK);
+
+            // the current context could already be completed with SUCCESS and we're just waiting for another
+            // one to show up while a timeout occurs or the channel closes. in these cases, this would be a valid
+            // close in all likelihood so there's no reason to log or alert the client as the client already has
+            // the best answer
+            if (!currentGremlinContext.isFinalResponseWritten()) {
+                logger.warn(rexex.getMessage(), rexex);
+                currentGremlinContext.writeAndFlush(rexex.getResponseMessage());
+            }
+        } finally {
+            // if this is a normal end to the session or if the session life timeout is exceeded then the
+            // session needs to be removed and everything cleaned up
+            if (closeReason.compareAndSet(null, CloseReason.NORMAL) || closeReason.get() == CloseReason.SESSION_TIMEOUT) {
+                close();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
```

Comment:
I'll add the javadoc. The call is mostly async as it basically just interrupts the thread running the `Rexster` instance and it is that gremlinPool thread that finishes its own cleanup. if this method was called by the gremlinPool thread it would be pretty fast to finish because there's nothing heavy left to do.